### PR TITLE
test(dist): add smoke test with github action

### PR DESCRIPTION
**Pull request by menski**
## Description

- add smoke-test.yaml for github action running on Windows, Linux and MacOS
- add smoke-test profile to dist module
- add SmokeTestIT test case to verify simple test case

### CODE REVIEW

File: dist/pom.xml [hunk 1]
```diff
@@ -247,5 +253,56 @@
...
+              <!-- prevent forking otherwise zeebe test listner is not found in classpath -->
+              <forkMode>never</forkMode>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>com.bazaarvoice.maven.plugins</groupId>
+            <artifactId>process-exec-maven-plugin</artifactId>
+            <version>0.7</version>
+            <executions>
+                <!--Start process broker-->
```
Comment (npepinpe) [inline dist/pom.xml:274]: Just a very quick note: you can easily spin up containers via Testcontainers. So as long as we've built the image then there shouldn't be any issue with running a normal container based test.

**Comment by npepinpe** [comment]
Seeing as it's from your own fork, I guess I can't quite take this over, but I'll use this as a baseline. Thanks! Enjoy your holidays!

File: dist/pom.xml (hunk 1 above)
Comment (menski) [reply dist/pom.xml:274]: Running Containers doesn't really make sense here as we want to check that the broker runs on the different operating systems

Comment (npepinpe) [reply dist/pom.xml:274]: That's true, don't know what I was thinking :smile:

**Comment by npepinpe** [comment]
Opened https://github.com/camunda-cloud/zeebe/pull/7658. This is heavily based on this PR, but the smoke tests are a little different. The Github workflow is the same, and the idea of the smoke test profile as well. The differences are:

- The tests can be ran from the IDE as well, without requiring a broker to be running before hand.
- The tests use the Spring Boot integration test framework to start the Spring Boot application in-JVM. This is beneficial in general for us as we had no tests (other than container based tests) which tests the Spring Boot envelope, i.e. `StandaloneBroker` and `StandaloneGateway`.
- Specific tests are annotated with a `@SmokeTest` (which is just a type-safe alias for `@Tag("smoke-test")` which is what the `smoke-test` profile will pick up.

Closing this then. Thanks for the help!

**Pull request closed without merging**